### DISCUSSION TITLE
Enabled Power Control vs Enabled Battery Control

### DISCUSSION THREAD

**Discussion by neildhunter** [wills106/homeassistant-solax-modbus#586] [Q&A] [answered]
Hopefully a quick and easy question(!) - but is there any difference between using Power vs. Battery control for the Modbus power control when trying to achieve a forced charge of the battery?

The descriptions seem to be similar (though not identical) so I'm wondering if there's a subtle difference?

**Comment by infradom** [comment]
Sorry for the confusing description in the wiki. Enabled_power_control is the very basic solax implementation where you specify the active power as such. Active power is notion that is difficult to understand (see wiki for a more complete explanation)

Conclusion: we can look at Activepower in 2 ways (both ways are correct):

- Activepower = grid(_import) - house_load
- Activepower = battery(_charge) - PV

In Enabled_battery_control mode, the active power is the specified target minus the current PV power.
In Enabled_power_control, the active power is the specified target without modification
In Enabled_grid_control, the active power is  the target - houseload

Confusing... but the name battery_control and grid_control try to explain what we try to reach.
I myself never use power_control directly, since it confuses me too much ...

**Comment by infradom** [comment, marked as answer]
So for a forced charge of the battery, no matter whether there is solar power or not, use battery_control.
Only if there is no PV power, both are the same ...

**Reply by neildhunter** [reply]
Thanks - not sure I follow the descriptions still but looks like it's battery_control that will do what I need to in this case!

**Reply by Tangiers2020** [reply]
sorry to bring this one back form the dead after a year! would you mind providing the numbers you are setting on that script please?
I am really struggling to get this right to charge form grid.

**Reply by neildhunter** [reply]
So for select.solax_remotecontrol_power_control I'm using "Enabled Battery Control" as the option and then in number.solax_remotecontrol_active_power I'm setting a target value for the battery. I'm just using 6000 as a figure there to let it find its own limit.
The only other value in my script sets the duration to 30 minutes through number.solax_remotecontrol_autorepeat_duration=1800.

**Comment by johny-mnemonic** [comment]
@infradom would you advice how to use Power Control to stop battery charging or even start discharging the battery, when there is PV power available?

When there is no PV it seems to work as expected, but I would like to limit the amount of time my battery is sitting charged to 100%, so I would like to stop battery charging e.g. on 80 or 90% even when there is enough PV to both charge the battery and export to the grid with full allowed power.

When I try to set it to "Enabled_battery_control" mode with "Activepower = 0" it still charges the battery from PV. And even setting some ridiculously high value like "Activepower = -5000" still doesn't cause it to stop charging or start discharging 😞

**Reply by johny-mnemonic** [reply]
Seems like I have found a workaround till we find a better solution (it writes to ROM, which is not ideal). 
Setting `number.solax_battery_charge_max_current` to 0 when you want to stop battery charging seems to work fine.
